src/main/java/MisterDuke.java

```diff
@@ -0,0 +1,678 @@
+                String inst = strArray[0].substring(0, strArray[0].length() - 2);
+                String date = strArray[1].substring(0, strArray[1].length() - 1).trim();
+                try {
+                    Date deadlineDate = new SimpleDateFormat("E MMM dd HH:mm:ss Z yyyy").parse(date);
+                    Deadline deadlineTask = new Deadline(inst + " ", deadlineDate);
+                    if (!checkForTick.contains(" ")) {
+                        deadlineTask.markAsDone();
+                    }
+                    tasksArrayList.add(deadlineTask);
+                } catch (Exception e) {
+                    e.getStackTrace();
+                }
+            } else if (tempStr.contains("[E]")) {
+                String[] strArray = cmd.split("at:", 2);
+                String inst = strArray[0].substring(0, strArray[0].length() - 2);
+                String date = strArray[1].substring(0, strArray[1].length() - 1);
+                Event eventTask = new Event(inst + " ", date);
+                if (!checkForTick.contains(" ")) {
+                    eventTask.markAsDone();
+                }
+                tasksArrayList.add(eventTask);
+            }
+        }
+        sc.close();
+        return tasksArrayList;
+    }
+
+    /**
+     * Saves the list of tasks on to the hard disk after Mister Duke
+     * has been terminated
+     * @param taskArrayList list of tasks
+     * @throws IOException when there is a wrong/incomplete user input
+     */
+    public void save(ArrayList<Task> taskArrayList) throws IOException {
+        FileWriter fwriter = new FileWriter(this.txtFile);
+        for (Task task : taskArrayList) {
+            if (task instanceof ToDo) {
+                fwriter.write(task.toString() + "\n");
+            } else if (task instanceof Deadline) {
+                fwriter.write(task.toString() + "\n");
+            } else {
+                fwriter.write(task.toString() + "\n");
+            }
+        }
+        fwriter.close();
+    }
+}
+
+/**
+ * The Parser class takes in the user input, parses it,
+ * then display the corresponding message from Mister Duke
+ */
+class Parser {
+    private final Ui ui;
+    private Storage storage;
+    private ArrayList<Task> taskList;
+
+    public Parser(Ui ui, Storage storage, ArrayList<Task> taskList) throws DukeException {
+        this.ui = ui;
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    /**
+     * The parse function takes in the user's input (command) and
+     * returns a class that corresponds to the command
+     * @param command user input
+     * @return class that corresponds to the command
+     * @throws DukeException when the user input is wrong/incomplete
+     */
+    public static Command parse(String command) throws DukeException {
+        String input = command.trim();
+        String[] strArray = input.split(" ", 2);
+        String cmd = strArray[0].trim();
+
+        if (cmd.equalsIgnoreCase("todo")) {
```

Comment:
Seeing that there are multiple if else cases, perhaps you may want to consider using a switch case? That would improve future developers to be able to figure out components easily by looking at the case conditions! 😃